From 8546561f631beac89e90fb9842dea5e0005ffd79 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 16 Aug 2018 17:47:33 +0300
Subject: [PATCH] scripts [flash_burn]: add verification

If next id is not "31313131" and not "32323232" burn both sides
I assume we got curropted or old running FW
---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn b/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
index ea59d88..6a0e782 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
@@ -63,12 +63,15 @@
 # 3 -- Uboot Env
 # 4 -- FW 0
 # 5 -- FW 1
-STABLE_IMAGE_CHAR=`xxd -p -l1 -c1 /dev/mtd2`
+STABLE_IMAGE_MAGIC=`xxd -p -l4 -c4 /dev/mtd2`
+STABLE_IMAGE_CHAR=${STABLE_IMAGE_MAGIC:0:2}
 # '1' / '2' equal (ascii) 0x31 / 0x32
-if [ ! "${STABLE_IMAGE_CHAR}" = "31" ] && [ ! "${STABLE_IMAGE_CHAR}" = "32" ]; then
+if [ "${STABLE_IMAGE_MAGIC}" != "31313131" ] && [ "${STABLE_IMAGE_MAGIC}" != "32323232" ]; then
+	STABLE_IMAGE_UNKNOWN=y
 	STABLE_IMAGE_CHAR=31
 fi
 STABLE_IMAGE_NUM=$((0x${STABLE_IMAGE_CHAR} - 0x31))
+STABLE_MTD_ID=$((4 + ${STABLE_IMAGE_NUM}))
 MODIFIABLE_IMAGE_ID=$((1 - ${STABLE_IMAGE_NUM}))
 MODIFIABLE_MTD_ID=$((4 + ${MODIFIABLE_IMAGE_ID}))
 BTL_IMAGE=${INFILE}.mtd1
@@ -120,3 +123,7 @@
 burn_image 2
 burn_image 3
 burn_image ${MODIFIABLE_MTD_ID}
+if [ "${STABLE_IMAGE_UNKNOWN}" == "y" ]; then
+	cp ${FW_IMAGE} ${INFILE}.mtd${STABLE_MTD_ID}
+	burn_image ${STABLE_MTD_ID}
+fi
